### PR TITLE
`should_record` fixup

### PR DESCRIPTION
**Pull request by Dexterp37**
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: glean-core/src/metrics/mod.rs [hunk 1]
```diff
@@ -22,11 +22,11 @@ pub use self::uuid::UuidMetric;
 
 #[derive(Serialize, Deserialize, Debug, Clone)]
 pub enum Metric {
-    String(String),
     Boolean(bool),
     Counter(i32),
-    Uuid(String),
+    String(String),
     StringList(Vec<String>),
+    Uuid(String),
```
Comment (badboy) [inline glean-core/src/metrics/mod.rs]: Just for ... alphabetic reasons?

File: glean-core/src/metrics/string_list.rs [hunk 2]
```diff
@@ -17,13 +18,19 @@ pub struct StringListMetric {
     meta: CommonMetricData,
 }
 
+impl MetricType for StringListMetric {
```
Comment (badboy) [inline glean-core/src/metrics/string_list.rs]: He, I explicitly left these untouched for whoever fully implements these types.
I don't mind getting them fixed now though.

File: glean-core/src/metrics/mod.rs (hunk 1 above)
Comment (Dexterp37) [reply glean-core/src/metrics/mod.rs]: Mh, yes, feels weird to have a random order there

File: glean-core/src/metrics/string_list.rs (hunk 2 above)
Comment (Dexterp37) [reply glean-core/src/metrics/string_list.rs]: Yes, I thought about that too, but then I figured that I was copy/pasting outdated code. I think we should either keep things up-to-speed or remove partial implementations. We can have these bits there, the rest is mostly tests and logic that the implementor would need to do anyway.

**Pull request merged**